### PR TITLE
feat(Wcostream): add presence

### LINKED ISSUE #5413: Wcostream | www.wcostream.com
### Discussed in https://github.com/PreMiD/Presences/discussions/4510

<div type='discussions-op-text'>

<sup>Originally posted by **aragocz** November  1, 2020</sup>
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** *Furuhashi Fumino#8496* / *ID : 428984613935775765*

**What's the name of the service?** https://www.wcostream.com/ | Wcostream

**What should the Presence display?** *Anime you're watching, episode, time left, etc.*

**If possible, please provide a logo for the service (512x512 minimum)** i couldn't find the size, you're looking for, but i got an icon from their webpage.
Here

</div>

### LINKED ISSUE #5682: Watch Cartoons Online | www.wcostream.com
### Website name

Watch Cartoons Online

### Website URL

https://www.wcostream.com/

### Website logo

https://www.wcostream.com/wp-content/themes/animewp78712/images/logo.gif

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

Title of show and/or episode.

Bonus cookie for how long until the episode is finished is displayed too.

### PR DESCRIPTION
**Pull request by kyrie25**
**Describe the changes in this pull request:**
Resolves #5413
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

image
image
image
image
image
image
image

</details>

### CODE REVIEW

File: websites/W/Watch Cartoon Online/presence.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
+const presence = new Presence({
+		clientId: "936985014560755753"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let video = {
+	timeLeft: "",
```
Comment (EncryptedDev) [inline websites/W/Watch Cartoon Online/presence.ts:7]: Shouldn't this be a number conventionally?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/W/Watch Cartoon Online/presence.ts (hunk 1 above)
Comment (kyrie25) [reply websites/W/Watch Cartoon Online/presence.ts:7]: It's using the format `MM:SS` and couldn't use as type `number`
~~I've tried several times with this and the presence I made before this, it just doesn't work somehow~~

Comment (kyrie25) [reply websites/W/Watch Cartoon Online/presence.ts:7]: image
Screenshot showing `endTimestamp` won't work if used as type `number`

File: websites/W/Watch Cartoon Online/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.6",
+	"author": {
+		"name": "Kyrie",
+		"id": "368399721494216706"
+	},
+	"service": "Watch Cartoon Online",
```
Comment (Slowlife01) [inline websites/W/Watch Cartoon Online/dist/metadata.json:7]: Should be `Wcostream`

Comment (kyrie25) [reply websites/W/Watch Cartoon Online/dist/metadata.json:7]: Done

File: websites/W/Wcostream/presence.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+			details: "Browsing...",
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname } = document.location,
+		[timestamps, cover, buttons] = await Promise.all([
+			presence.getSetting<boolean>("timestamps"),
+			presence.getSetting<boolean>("cover"),
+			presence.getSetting<boolean>("buttons")
+		]);
+	if (video?.timeLeft !== "") {
```
Comment (Bas950) [inline websites/W/Wcostream/presence.ts:27]: pretty sure video is always defined.

Comment (kyrie25) [reply websites/W/Wcostream/presence.ts:27]: Yeah that makes sense

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**